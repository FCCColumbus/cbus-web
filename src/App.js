import './App.css';
import Header from './components/Header';
import Splash from './components/Splash';
import About from './components/About';
import Events from './components/Events'
import Members from './components/Members';
import Footer from './components/Footer';

function App() {
  return (
    <div className="App">
      <Header />
      <Splash />
      <About />
      <Events />
      <img id="divider" src='/divider.jpg' alt='code on a screen. mostly css' />
      <Members />
      <Footer />
    </div>
  );
}

export default App;
