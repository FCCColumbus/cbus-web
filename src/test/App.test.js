import { render } from "@testing-library/react";

import App from "../App";

// The mockName isn't necessary but may be helpful for reading test failures.
const randomString = (mockName) => `${mockName}_${Math.random().toString(36).substring(2, 8)}`;

const mockHeader = randomString("Header");
const mockSplash = randomString("Splash");
const mockAbout = randomString("About");
const mockEvents = randomString("Events");
const mockDividerImage = randomString("DividerImage");
const mockMembers = randomString("Members");
const mockFooter = randomString("Footer");

jest.mock('../components/Header', () => () => mockHeader);
jest.mock('../components/Splash', () => () => mockSplash);
jest.mock('../components/About', () => () => mockAbout);
jest.mock('../components/Events', () => () => mockEvents);
jest.mock('../components/DividerImage', () => () => mockDividerImage);
jest.mock('../components/Members', () => () => mockMembers);
jest.mock('../components/Footer', () => () => mockFooter);

describe("App", () => {
  test("renders App components in order", () => {
    // Capture the raw output of the render function.
    // It returns a data type called "RenderResult", so that's what we'll name the variable.
    const renderResult = render(<App />);

    // Isolate the thing that we want to test.
    // I usually name this variable "actual" as it represents what our code is actually doing,
    // and is consistent with the "actual-expected" terminology used in unit testing.
    // In this case, our RenderResult object has a property called "container" and we want to test
    // the innerHTML of that container.
    // In the context of React/JSX, the innerHTML is what's generated by the component's return statement.
    const actual = renderResult.container.innerHTML;

    const expected = `<div class="App">${mockHeader}${mockSplash}${mockAbout}${mockEvents}${mockDividerImage}${mockMembers}${mockFooter}</div>`;
    expect(actual).toBe(expected)
  });
});
