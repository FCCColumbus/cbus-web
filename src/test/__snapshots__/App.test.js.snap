// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <header>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            <img
              alt="fcc logo"
              className="header-logo"
              src="/fcc-logo.svg"
            />
          </a>
        </li>
        <li>
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="https://www.freecodecamp.org/"
          >
            FreeCodeCamp
          </a>
        </li>
        <li>
          <a
            href="https://discord.com/invite/EXehPVnBYz"
          >
            Discord
          </a>
        </li>
        <li>
          <a
            href="https://github.com/FCCColumbus"
          >
            GitHub
          </a>
        </li>
        <li>
          <a
            href="#events"
          >
            Events
          </a>
        </li>
        <li>
          <a
            href="https://github.com/FCCColumbus/cbus-web/graphs/contributors"
          >
            Contributors
          </a>
        </li>
      </ul>
    </nav>
    <h1>
      free code camp
      <img
        alt="fcc logo"
        className="big-logo"
        src="free-code-camp-logo.svg"
      />
       columbus
    </h1>
  </header>
  <div
    className="splash"
  >
    <div
      className="byline"
    >
      <h2>
        Free Code Camp is the local chapter of FCC, the best way to learn coding for free!
      </h2>
    </div>
    <img
      alt="people having fun coding"
      src="/cottonbro-studio.jpg"
    />
  </div>
  <div
    className="about"
    id="about"
  >
    <h2>
      About
    </h2>
    <p>
      Whether you are a beginner, or a pro, there is always something new to learn, or someone new to meet.
    </p>
    <h3>
      Here is why you should come
    </h3>
    <div
      className="center"
    >
      <ul
        className="come"
      >
        <li>
          <b>
            ☛
          </b>
           Get help in real time from our community Discord page.
        </li>
        <li>
          <b>
            ☛
          </b>
           Meet up with other techie Columbusites.
        </li>
        <li>
          <b>
            ☛
          </b>
           Learn to code at your own pace, in your browser or on your phone.
        </li>
        <li>
          <b>
            ☛
          </b>
           Work through our focused, interactive courses and tutorials.
        </li>
        <li>
          <b>
            ☛
          </b>
           Learn state-of-the-art full stack JavaScript tutorials.
        </li>
        <li>
          <b>
            ☛
          </b>
           Get real-world experience building projects that help local nonprofits.
        </li>
        <li>
          <b>
            ☛
          </b>
           Get a free certificate of completion.
        </li>
        <li>
          <b>
            ☛
          </b>
           Assemble a portfolio of real apps used by real people.
        </li>
      </ul>
    </div>
  </div>
  <div
    className="events"
    id="events"
  >
    <h3>
      Upcoming Events
    </h3>
    <div
      id="events-placeholder-text"
    >
      The "Events" section is under development.
      <br />
      <br />
      In the meantime, please visit
      <br />
      <a
        href="https://www.meetup.com/techlifecolumbus/events/"
      >
        https://www.meetup.com/techlifecolumbus/events/
      </a>
      <br />
      and look for FreeCodeCamp Columbus events there!
    </div>
  </div>
  <img
    alt="code on a screen. mostly css"
    src="/divider.jpg"
  />
  <div>
    <div
      className="members"
    >
      <h3>
        Meet the Admins
      </h3>
      <div
        className="row"
      >
        <div>
          <h4>
            John
            <br />
             
            Riley
          </h4>
          <img
            alt="beautiful headshot of John Riley."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/41979303?v=4"
          />
          <h5>
            Admin & Coding Mentor
          </h5>
        </div>
        <div>
          <h4>
            Kent
            <br />
             
            Utterback
          </h4>
          <img
            alt="beautiful headshot of Kent Utterback."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/17324971?v=4"
          />
          <h5>
            Admin Emeritus & Coding Mentor
          </h5>
        </div>
        <div>
          <h4>
            Amy
            <br />
             
            Bedinghaus
          </h4>
          <img
            alt="beautiful headshot of Amy Bedinghaus."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/28076677?v=4"
          />
          <h5>
            Admin Emeritus
          </h5>
        </div>
        <div>
          <h4>
            Austin
            <br />
             
            Truss
          </h4>
          <img
            alt="beautiful headshot of Austin Truss."
            className="avatar admin-avatar"
            src="https://media.licdn.com/dms/image/C5603AQGQpynBhWiZLA/profile-displayphoto-shrink_200_200/0/1517355585710?e=1696464000&v=beta&t=DrPEzm5U1we4G7qUPoFrAA5gw_MHSN3dU8_02iWZYRA"
          />
          <h5>
            Admin Emeritus
          </h5>
        </div>
        <div>
          <h4>
            Kevin
            <br />
             
            Bruland
          </h4>
          <img
            alt="beautiful headshot of Kevin Bruland."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/17850505?v=4"
          />
          <h5>
            Admin Emeritus
          </h5>
        </div>
        <div>
          <h4>
            Sarah
            <br />
             
            Lilly-Bruland
          </h4>
          <img
            alt="beautiful headshot of Sarah Lilly-Bruland."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/17872422?v=4"
          />
          <h5>
            Admin Emeritus
          </h5>
        </div>
        <div>
          <h4>
            Justin
            <br />
             
            Woodward
          </h4>
          <img
            alt="beautiful headshot of Justin Woodward."
            className="avatar admin-avatar"
            src="https://avatars.githubusercontent.com/u/17994108?v=4"
          />
          <h5>
            Admin Emeritus
          </h5>
        </div>
      </div>
    </div>
    <div
      className="members"
    >
      <h3>
        Meet the Contributors
      </h3>
      <div
        className="row"
      >
        <div>
          <img
            alt="beautiful headshot of John."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            John
          </h4>
        </div>
        <div>
          <img
            alt="beautiful headshot of Ismael."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            Ismael
          </h4>
        </div>
        <div>
          <img
            alt="beautiful headshot of Nirvignesh."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            Nirvignesh
          </h4>
        </div>
        <div>
          <img
            alt="beautiful headshot of Tijana."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            Tijana
          </h4>
        </div>
        <div>
          <img
            alt="beautiful headshot of Jennifer."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            Jennifer
          </h4>
        </div>
        <div>
          <img
            alt="beautiful headshot of Kellan."
            className="avatar member-avatar"
            src="/head.png"
          />
          <h4>
            Kellan
          </h4>
        </div>
      </div>
    </div>
  </div>
  <footer>
    <nav>
      <ul>
        <li>
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="https://www.freecodecamp.org/"
          >
            FreeCodeCamp
          </a>
        </li>
        <li>
          <a
            href="https://discord.com/invite/EXehPVnBYz"
          >
            Discord
          </a>
        </li>
        <li>
          <a
            href="https://github.com/FCCColumbus"
          >
            GitHub
          </a>
        </li>
        <li>
          <a
            href="#events"
          >
            Events
          </a>
        </li>
        <li>
          <a
            href="https://github.com/FCCColumbus/cbus-web/graphs/contributors"
          >
            Contributors
          </a>
        </li>
      </ul>
    </nav>
    <p>
      Copyright Free Code Camp Columbus 2023 - 
      2023
    </p>
    <a
      className="image-wrapper"
      href="https://www.netlify.com"
      rel="noreferrer"
      target="_blank"
    >
       
      <img
        alt="Deploys by Netlify"
        src="https://www.netlify.com/v3/img/components/netlify-color-accent.svg"
      />
       
    </a>
  </footer>
</div>
`;
